packages/grafana-ui/src/components/TagsInput/TagsInput.tsx

```diff
@@ -1,122 +1,88 @@
-import React, { ChangeEvent, KeyboardEvent, PureComponent } from 'react';
-import { css, cx } from 'emotion';
-import { stylesFactory } from '../../themes/stylesFactory';
+import React, { ChangeEvent, KeyboardEvent, FC, useState } from 'react';
+import { css } from 'emotion';
 import { Button } from '../Button';
-import { Input } from '../Forms/Legacy/Input/Input';
 import { TagItem } from './TagItem';
+import { useStyles } from '../../themes/ThemeContext';
+import { GrafanaTheme } from '@grafana/data';
+import { Input } from '../Input/Input';
 
-interface Props {
+export interface Props {
   placeholder?: string;
   tags?: string[];
-
   onChange: (tags: string[]) => void;
 }
 
-interface State {
-  newTag: string;
-  tags: string[];
-}
-
-export class TagsInput extends PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    this.state = {
-      newTag: '',
-      tags: this.props.tags || [],
-    };
-  }
+export const TagsInput: FC<Props> = ({ placeholder = 'New tag (enter key to add)', tags = [], onChange }) => {
+  const [newTagName, setNewName] = useState<string>('');
```

Comment:
I think the types are inferred from default value, so need to specify them. 
```suggestion
  const [newTagName, setNewName] = useState('');
```